### PR TITLE
Fix inspector popup

### PR DESCRIPTION
**Pull request by cichli**

### CODE REVIEW

**Comment by cichli** [comment]
This might actually be a bug in `cider-popup-buffer-display`. If the buffer it's trying to pop up is already selected, maybe it should be a no-op.

**Comment by bbatsov** [comment]
Can you describe the problem this fixes in a bit more detail?

> This might actually be a bug in cider-popup-buffer-display. If the buffer it's trying to pop up is already selected, maybe it should be a no-op.

This sounds sensible. I recall @vspinu worked a few months ago on some popup buffer improvements, though I can't recall if they were related. I'll have to consult the code and the history.

**Comment by vspinu** [comment]
>> This might actually be a bug in cider-popup-buffer-display. If the buffer it's trying to pop up is already selected, maybe it should be a no-op.

>This sounds sensible. 

Yes it does. I think I missed this pattern because it's a "popup-up" buffer.

**Comment by cichli** [comment]
@bbatsov Sorry, I should have been more clear. It's easy to replicate.

* `M-x` `cider-inspect` and eval a form. The inspector buffer will pop up in a new window.
* Perform any inspector operation in the buffer, e.g. refresh it with `g` or inspect a value with `RET`. The inspector buffer will now pop up in the window you previously had selected.
* Repeat the second step and the inspector continues to pop up in another window.

@vspinu thanks for clarifying. I've just pushed again with the fix applied to `cider-popup-buffer-display`.

File: cider-interaction.el [hunk 1]
```diff
@@ -1196,13 +1196,14 @@ If major MODE is non-nil, enable it for the popup buffer."
...
   "Display BUFFER.
 If SELECT is non-nil, select the BUFFER."
-  (-when-let (win (get-buffer-window buffer))
-    (with-current-buffer buffer
-      (set-window-point win (point))))
-  ;; Non nil `inhibit-same-window' ensures that current window is not covered
-  (if select
-      (pop-to-buffer buffer `(nil . ((inhibit-same-window . ,pop-up-windows))))
-    (display-buffer buffer `(nil . ((inhibit-same-window . ,pop-up-windows)))))
+  (when (not (eq buffer (current-buffer)))
```
Comment (bbatsov) [inline cider-interaction.el]: when + not == unless

Comment (cichli) [inline cider-interaction.el]: Thanks; fixed

File: cider-interaction.el [hunk 2]
```diff
@@ -1196,13 +1196,14 @@ If major MODE is non-nil, enable it for the popup buffer."
...
   "Display BUFFER.
 If SELECT is non-nil, select the BUFFER."
-  (-when-let (win (get-buffer-window buffer))
-    (with-current-buffer buffer
-      (set-window-point win (point))))
-  ;; Non nil `inhibit-same-window' ensures that current window is not covered
-  (if select
-      (pop-to-buffer buffer `(nil . ((inhibit-same-window . ,pop-up-windows))))
-    (display-buffer buffer `(nil . ((inhibit-same-window . ,pop-up-windows)))))
+  (unless (eq buffer (current-buffer))
```
Comment (vspinu) [inline cider-interaction.el]: It's safer to use `(eq (get-buffer buffer) (current-buffer))` as buffer might be buffer name. All emacs functions that accept a buffer also accept a string.

Comment (cichli) [inline cider-interaction.el]: @vspinu I've fixed this too, thanks!

**Comment by bbatsov** [comment]
:+1: Thanks!

**Pull request merged**